lls/main.c

```diff
@@ -16,15 +16,36 @@
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
+#include <stdbool.h>
+
 #include <rte_cycles.h>
 #include <rte_ethdev.h>
 
+#include "arp.h"
 #include "cache.h"
 
 /* Length of time (in seconds) to wait between scans of the cache. */
 #define LLS_CACHE_SCAN_INTERVAL 10
 
-static struct lls_config lls_conf;
+static struct lls_config lls_conf = {
+	.arp_cache = {
+		.key_len = sizeof(struct in_addr),
+		.key_str_len = INET_ADDRSTRLEN,
+		.name = "arp",
+		.iface_enabled = arp_iface_enabled,
+		.entry_str = ipv4_str,
+		.xmit_req = xmit_arp_req,
+		.print_record = print_arp_record,
+	},
+	/* TODO Initialize ND cache. */
+};
+
+static inline int
+arp_enabled(struct lls_config *lls_conf)
+{
+	return lls_conf->arp_cache.iface_enabled(&lls_conf->net->front) ||
+		lls_conf->arp_cache.iface_enabled(&lls_conf->net->back);
```

Comment:
Since I merged the back interface check into `iface_enabled` per the suggestion above, this removes the need to add this check to all of these cases in lls/main.c.